mindsdb/integrations/handlers/google_gemini_handler/google_gemini_handler.py

```diff
@@ -0,0 +1,102 @@
+import os
+from typing import Dict, Optional
+
+import google.generativeai as genai
+import pandas as pd
+from mindsdb.integrations.libs.base import BaseMLEngine
+from mindsdb.utilities import log
+from mindsdb.utilities.config import Config
+
+logger = log.getLogger(__name__)
+
+
+class GoogleGeminiHandler(BaseMLEngine):
+    """
+    Integration with the Google generative AI Python Library
+    """
+
+    name = "google_gemini"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.default_chat_model = "gemini-pro"
+        self.supported_chat_models = ["gemini-pro"]
+        self.generative = True
+        self.connection = None
+
+    def create(
+        self,
+        target: str,
+        df: Optional[pd.DataFrame] = None,
+        args: Optional[Dict] = None,
+    ) -> None:
+        if "model" not in args["using"]:
+            args["using"]["model"] = self.default_chat_model
+        elif args["using"]["model"] not in self.supported_chat_models:
+            raise Exception(
+                f"Invalid chat model. Please use one of {self.supported_chat_models}"
```

Comment:
May be worth checking api key is valid in create

Reply:
yes sure